src/app/guard/auth.guard.ts

```diff
@@ -1,15 +1,27 @@
-import { Injectable } from '@angular/core';
-import { ActivatedRouteSnapshot, CanActivate, RouterStateSnapshot, UrlTree } from '@angular/router';
-import { Observable } from 'rxjs';
+import {Injectable} from '@angular/core';
+import {ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree} from '@angular/router';
+import {Observable} from 'rxjs';
+import {AuthService} from "../user/auth.service";
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthGuard implements CanActivate {
+  constructor(private authService: AuthService, private router: Router) {
+  }
+
   canActivate(
     route: ActivatedRouteSnapshot,
     state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
-    return true;
+    return this.checkLoggedIn(state.url);
+  }
+
+  checkLoggedIn(url: string): boolean {
+    if (this.authService.isLoggedIn) {
+      return true;
+    }
+    this.authService.redirectUrl = url;
+    this.router.navigate(['/welcome']);
+    return false;
```

Comment:
can you refactor this a bit? you can return the flag (this.authService.isLoggedIn) instead of true or false...